packages/ibm-products/src/components/ConditionBuilder/ConditionGroupBuilder/ConditionGroupBuilder.js

```diff
@@ -45,12 +46,46 @@ const ConditionGroupBuilder = ({
       variant == 'tree'
         ? handleFocusOnCloseTree(evt)
         : handleFocusOnClose(evt, conditionIndex);
-      onChange({
-        ...group,
-        conditions: group.conditions.filter(
-          (condition) => conditionId !== condition.id
-        ),
-      });
+
+      if (group.conditions[1].conditions) {
+        //when we remove every plain conditions of a group without deleting  the subgroup, we need to restructure the group
+        //the inner group become outer group and same level subgroups become plain conditions
+
+        let newGroup = [],
+          allConditions = [];
+        group.conditions.map((eachItem) => {
+          if (eachItem.conditions) {
+            allConditions = [...allConditions, ...eachItem.conditions];
+          } //spreading out the condition inside the subgroup
+        });
+        //we always have conditions first and then subgroups, so ordering accordingly
+        const groupedItems = allConditions.reduce((acc, item) => {
+          if (item.conditions) {
+            acc['groups'] = acc['groups'] || [];
+            acc['groups'].push(item);
+          } else {
+            acc['conditions'] = acc['conditions'] || [];
+            acc['conditions'].push(item);
+          }
+          return acc;
+        }, {});
+
+        newGroup.conditions = [
```

Comment:
if i'm reading this correctly it looks like this is just `allConditions`. if that's the case then `groupedItems` really inst necessary at all since you can just do `newGroup.conditions = [].concat(allConditions)`. although i would like to point out that you're using property assignment on an array here, which should most likely be done on an object instead.

Reply:
Actually its just `allConditions` . But allConditions will contain normal conditions as well as any subgroups. So we need rearrange `allConditions` in such a way that conditions first and then any subgroups. This is an edge scenario when user delete all conditions in a group. 

https://github.com/carbon-design-system/ibm-products/assets/153802538/4e1e3043-e79d-499e-8c70-512ffd151f92